source/server/lds_api.cc

```diff
@@ -16,6 +16,14 @@ LdsApi::LdsApi(const Json::Object& config, Upstream::ClusterManager& cm,
                      std::chrono::milliseconds(config.getInteger("refresh_delay_ms", 30000))),
       local_info_(local_info), listener_manager_(lm),
       stats_({ALL_LDS_STATS(POOL_COUNTER_PREFIX(scope, "listener_manager.lds."))}) {
+  if (local_info.clusterName().empty() || local_info.nodeName().empty()) {
+    throw EnvoyException("lds: setting --service-cluster and --service-node are required");
```

Comment:
I notice we do this in other places, like CDS too. Why not refactor to push this check much earlier to some common path?